pandas/io/parquet.py

```diff
@@ -224,7 +235,8 @@ def write(self, df, path, compression='snappy', index=None, **kwargs):
 
         with catch_warnings(record=True):
             self.api.write(path, df, compression=compression,
-                           write_index=index, **kwargs)
+                           write_index=index, partition_on=partition_cols,
```

Comment:
What if a user does

```python
df.to_parquet("/tmp/a", engine="fastparquet", partition_on=['foo'])
```

I think that would raise an error, since you've specified `partition_on` twice. It should use the user-provided `partition_on`.

Reply:
i feel using one argument for partitioning makes more sense. Why don't we raise an error, asking them to use partition_cols instead in that case

Reply:
make sense.. will handle it such that partition_on gets more priority than partition_cols with a warning

Reply:
I'm not sure it should even be a warning. If a warning is going to be emitted, then that should be done by fastparquet.